Remove needsBlackFullscreenBackgroundQuirk

karlcow opened the pull request:
#### 6525018e7c182a59c811dfe4e5f37e606ebf89be
<pre>
Remove needsBlackFullscreenBackgroundQuirk
<a href="https://bugs.webkit.org/show_bug.cgi?id=248323">https://bugs.webkit.org/show_bug.cgi?id=248323</a>
rdar://102653160

Reviewed by Tim Nguyen.

This is removing the mlb.com quirk which was needed when
the background was white. Backdrop will take over.
A simple change to allow the fake backdrop to take effect.
SeeAlso <a href="https://github.com/WebKit/WebKit/pull/6021/files">https://github.com/WebKit/WebKit/pull/6021/files</a>
This also aligns WebKit with Gecko for the fullscreen
background.
Further fixes will be made in
<a href="https://github.com/WebKit/WebKit/pull/6688">https://github.com/WebKit/WebKit/pull/6688</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=248148">https://bugs.webkit.org/show_bug.cgi?id=248148</a>

* LayoutTests/TestExpectations:
* LayoutTests/compositing/no-compositing-when-fulll-screen-is-present-expected.txt:
* LayoutTests/platform/gtk/compositing/no-compositing-when-fulll-screen-is-present-expected.txt:
* Source/WebCore/css/fullscreen.css:
(:-webkit-full-screen):
* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::needsBlackFullscreenBackgroundQuirk const): Deleted.
* Source/WebCore/page/Quirks.h:
* Source/WebCore/rendering/RenderTheme.h:
(WebCore::RenderTheme::extraFullScreenStyleSheet): Deleted.
* Source/WebCore/style/UserAgentStyle.cpp:
(WebCore::Style::UserAgentStyle::ensureDefaultStyleSheetsForElement):

Canonical link: <a href="https://commits.webkit.org/257271@main">https://commits.webkit.org/257271@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/58ca6b24cb73dc0b5331f015741b2946a596b728

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/98411 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/7615 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/31538 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/107840 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/168109 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/102348 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/8126 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/84994 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/90963 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/104499 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/104066 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/6175 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/89736 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/33170 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/87988 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/21098 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/76116 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/1559 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/22626 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/82/builds/1489 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/45114 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/74/builds/5011 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/6405 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/42038 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2501 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/2856 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/style/UserAgentStyle.cpp:
```diff
@@ -234,8 +234,6 @@ void UserAgentStyle::ensureDefaultStyleSheetsForElement(const Element& element)
-        if (element.document().quirks().needsBlackFullscreenBackgroundQuirk())
-            fullscreenRules.append(":-webkit-full-screen { background-color: black; }"_s);
         fullscreenStyleSheet = parseUASheet(fullscreenRules.toString());
         addToDefaultStyle(*fullscreenStyleSheet);
```
Since `extraFullscreenStyleSheet()` is always an empty string (see: https://searchfox.org/wubkat/search?q=extraFullScreenStyleSheet&path=&case=false&regexp=false), we can simplify this and this can become:
```
fullscreenStyleSheet = parseUASheet(StringImpl::createWithoutCopying(fullscreenUserAgentStyleSheet, sizeof(fullscreenUserAgentStyleSheet)));
addToDefaultStyle(*fullscreenStyleSheet);
```

karlcow replied:
I will do a cycle of recompile. Might take a bit of time.

karlcow replied:
ok this is working locally in my own build.

karlcow replied:
OK repushing.
Modified the line as you mentioned. Thanks.

karlcow replied:
5th time the charm 😄 pushed.

nt1m reviewed (commented):
Looks fine assuming you've tested this manually. Might be worth mentioning that this puts us in-line with Firefox in the commit message.

nt1m commented on Source/WebCore/style/UserAgentStyle.cpp:
```diff
@@ -234,9 +234,7 @@ void UserAgentStyle::ensureDefaultStyleSheetsForElement(const Element& element)
         fullscreenRules.append(RenderTheme::singleton().extraFullScreenStyleSheet());
-        if (element.document().quirks().needsBlackFullscreenBackgroundQuirk())
-            fullscreenRules.append(":-webkit-full-screen { background-color: black; }"_s);
-        fullscreenStyleSheet = parseUASheet(fullscreenRules.toString());
```
Sorry I wasn't clear, all these lines can be removed.

And these 3 lines too: https://searchfox.org/wubkat/rev/fde0f31b8e7849e3eb1bf6c82439b4b9a5d31baf/Source/WebCore/rendering/RenderTheme.h#96-98

nt1m reviewed (commented):
2 more things:
- `#include "Quirks.h"` can be removed I think
- compositing/no-compositing-when-fulll-screen-is-present.html needs a rebaseline according to EWS (just one line to remove)

karlcow committed:
Remove needsBlackFullscreenBackgroundQuirk
https://bugs.webkit.org/show_bug.cgi?id=248323
rdar://102653160

Reviewed by Tim Nguyen.

This is removing the mlb.com quirk which was needed when
the background was white. Backdrop will take over.
A simple change to allow the fake backdrop to take effect.
SeeAlso https://github.com/WebKit/WebKit/pull/6021/files
This also aligns WebKit with Gecko for the fullscreen
background.
Further fixes will be made in
https://github.com/WebKit/WebKit/pull/6688
https://bugs.webkit.org/show_bug.cgi?id=248148

* LayoutTests/TestExpectations:
* LayoutTests/compositing/no-compositing-when-fulll-screen-is-present-expected.txt:
* LayoutTests/platform/gtk/compositing/no-compositing-when-fulll-screen-is-present-expected.txt:
* Source/WebCore/css/fullscreen.css:
(:-webkit-full-screen):
* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::needsBlackFullscreenBackgroundQuirk const): Deleted.
* Source/WebCore/page/Quirks.h:
* Source/WebCore/rendering/RenderTheme.h:
(WebCore::RenderTheme::extraFullScreenStyleSheet): Deleted.
* Source/WebCore/style/UserAgentStyle.cpp:
(WebCore::Style::UserAgentStyle::ensureDefaultStyleSheetsForElement):

Canonical link: https://commits.webkit.org/257271@main
